Describe copyRootClassesToWrapper (#2638)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxOverlay/1 Configuration/copyRootClassesToWrapper.md

---
---
##### shortDescription
Copies user-defined classes from the root element to the wrapper element. 

---

> The following markup illustrates relations between wrapper and root elements when you set **copyRootClassesToWrapper** to **true**:

Reviewer: In which cases this property should be enabled?

Reviewer 2: `relations` is not the right word. Do you mean 'relationship'?

    <!--HTML--><body>
        
        <!-- The following is the root element on which the component is initialized. -->
        <!-- Specify classes to be copied to the wrapper element here. -->

>         <div id="{widgetname}" class="custom-class" ... ></div>

Reviewer: AFAIK, we don't have a `{widgetname}` placeholder.

Author: We've already used it here: https://github.com/DevExpress/devextreme-documentation/blob/21_1/api-reference/10%20UI%20Components/GridBase/1%20Configuration/highlightChanges.md

![image](https://user-images.githubusercontent.com/33040934/127384664-fbe008de-beb5-442f-bea3-9adaaeaeefe3.png)

---
---
##### shortDescription
Copies your custom CSS classes from the root element to the wrapper element. 

---

> This property allows you to use CSS for wrapper element configuration. You can customize shade, resize, or drag zone.

Reviewer: ```suggestion
This property allows you to use CSS for wrapper element configuration. You can customize the shade, resize, and drag zones.
```

> This property allows you to use CSS for wrapper element configuration. You can customize the shade, resize, and drag zones.

Reviewer 2: ```suggestion
This property allows you to use CSS for wrapper element configuration. You can resize, drag, and customize the shade of zones.
```

Author: Hello @dirk-pieterse 

Seems like my text is not clear enough. The point is the following: we have three zones - the drag zone, the resize zone and the shade zone. Users can customize each one of these zones.

Reviewer 2: I misunderstood the initial meaning of your sentence. It didn't seem like different zones, rather different things you can do with a zone.

The following markup illustrates relations between wrapper and root elements when you set **copyRootClassesToWrapper** to **true**:

    <!--HTML--><body>
        

>         <!-- The following is the root element on which the component is initialized. -->

Reviewer 2: - `The following` The following what?
- `on which the component is initialized` How can something be initialized **on** something else?

Author: >The following what?

We have comments on line 18 and 19 that describe what's going on in the code on line 20. 'The following' refers to the `<div ... ></div>` element.

>How can something be initialized on something else

It is a common thing. We have an element in markup (`<div`) and a component is 
'attached' to it when the component is initialized. But when the component is used in the application, it is automatically 'attached' to some other element (usually `<body>`). However, my description of this process is not technically precise, and the right term is "the component is initialized on some element". 

You can also find similar constructions in other docs (for example, Adobe):
![image](https://user-images.githubusercontent.com/33040934/127837231-b7e68fc6-d056-47e5-ba07-8312ede8dc2f.png)
![image](https://user-images.githubusercontent.com/33040934/127837252-c3d4191a-ff24-4e1e-9f42-a9c0f06a6537.png)

Reviewer 2: 1) The 'following' is incomplete, whereas the 'following element' is complete
2) Yes, I've seen this before in Adobe and some other documentation, however, it doesn't mean it is correct. I've seen countless questionable phrases and wording in Microsoft docs too. But ok, if you feel comfortable with this wording, then use it.

        <!-- Specify classes to be copied to the wrapper element here. -->
        <div id="{widgetname}" class="custom-class" ... ></div>

        

>         <!-- The following is the wrapper element. -->

Reviewer 2: Same here

Author: Same as above - 

>'The following' refers to the <div ... ></div> element.

---
---
##### shortDescription

> Copies user-defined classes from the root element to the wrapper element. 

Reviewer: ```suggestion
Copies your custom CSS classes from the root element to the wrapper element. 
```

---

> This property allows you to configure a single {widgetName}'s wrapper element to customize shade, resize, or drag zone.

Reviewer: ```suggestion
This property allows you to use CSS for wrapper element configuration. You can customize shade, resize, or drag zone.
```

> customize shade, resize, or drag zone.

What does that mean?

Author: something like this:
![image](https://user-images.githubusercontent.com/33040934/127663129-a3170751-d659-497e-b1ee-4ed2ab0cfaef.png)

Reviewer: Okay, that's for shade customization. What about the rest ("resize, or drag zone")?

Author: Seems like nobody except Anton knows these scenarios. He definitely said that 'the wrapper is responsible for shade, resize and drag zone', but I cannot request clarification since he took a personal day.

The following markup illustrates relations between wrapper and root elements when you set **copyRootClassesToWrapper** to **true**:

    <!--HTML--><body>
        
        <!-- The following is the root element on which the component is initialized. -->
        <!-- Specify classes to be copied to the wrapper element here. -->
        <div id="{widgetname}" class="custom-class" ... ></div>

        
        <!-- The following is the wrapper element. -->

>         <!-- User-defined classes ("custom-class" in this example) are automatically copied from the root element. -->

Reviewer: ```suggestion
        <!-- Custom classes are automatically copied from the root element. -->
```
